### ISSUE TITLE
autocomplete pop-up not shown at original cursor when change occurrence

### ISSUE THREAD

**Issue by t9md** [t9md/atom-vim-mode-plus#554]
- keystroke: `c o f` at cursor is `newRanges`
- Expected: autocomplete popup shows at original cursor position
- Actual: popup shown at last occurrence

![last-cursor-placement-of-occurrence](https://cloud.githubusercontent.com/assets/155205/20861696/c8e532da-b9d9-11e6-8efa-3c83ea1931f1.gif)

**Comment by t9md** [comment]
- autocomplete-popup is shown only at last-cursor.
- currently when select `occurrence`, I use internally `editor.setSelectedBufferRanges(ranges)` which reset current last-selection.
- So fix is placing last selection at original cursor position.
- Following situation should be considered
  - When `preset-occurrence` marker is used. cursor position is not necessarily ON occurrence. Since user can move cursor after putting `preset-occurrence` marker

### [Feedback-welcome-idea] where to put last selection is determined in following order

Not sure if 2, 3 works well, will evaluate.

1. On occurrence which intersects original selection head position
2. On occurrence in forward direction within visible screen range.
3. On occurrence in backward direction within visible screen range.
4. On occurrence in forward direction in whole buffer, if it reached EOF, then continue from top of buffer till already searched line(top line of visible screen)

**Comment by t9md** [comment]
Done 🎉  this is really important improvement!

## Test case

Place cursor on `@editor` then `c o f abc` to transform `@editor` to `@abc` within function

## Before

- autocomplete popup was shown at bottom cursor position
- `escape` in `normal-mode` clear multi-cursors but remaining cursor was topmost one.

![before](https://cloud.githubusercontent.com/assets/155205/20868101/9a98386a-ba96-11e6-8c2f-45d9b1cfd3c4.gif)


## After

- autocomplete popup was shown at original cursor position.
- `escape` in `normal-mode` clear multi-cursors respect original cursor position

![after](https://cloud.githubusercontent.com/assets/155205/20868102/a173935a-ba96-11e6-8ded-f9689df46f84.gif)

**Comment by trkoch** [comment]
Welcome enhancement to default behavior! While fiddling with #899 I noticed this workaround (I suspect in `lib/occurence-manager.js:188`) does not seem to be applied to multiple cursors created with `cmd+d`, which in turn causes to reset to last cursor instead of first. Perhaps if we could make sure order of cursors is the same for both methods we could obviate the need for #1097 at its root? I noticed `cmd+d` does not seem to be bound to atom-vim?
